content/en/aws/kms/index.md

```diff
@@ -0,0 +1,123 @@
+        "KeyId": "010a4301-4205-4df8-ae52-4c2895d47326",
+        "Arn": "arn:aws:kms:us-east-1:000000000000:key/010a4301-4205-4df8-ae52-4c2895d47326",
+        "CreationDate": 1665432947.493433,
+        "Enabled": true,
+        "Description": "",
+        "KeyUsage": "ENCRYPT_DECRYPT",
+        "KeyState": "Enabled",
+        "Origin": "AWS_KMS",
+        "KeyManager": "CUSTOMER",
+        "CustomerMasterKeySpec": "SYMMETRIC_DEFAULT",
+        "KeySpec": "SYMMETRIC_DEFAULT",
+        "EncryptionAlgorithms": [
+            "SYMMETRIC_DEFAULT"
+        ],
+        "MultiRegion": false
+    }
+}
+```
+
+If we lose the ID, we can always use
+
+{{< command >}}
+$ awslocal kms list-keys
+{{</ command >}}
+
+to list IDs and [Amazon Resource Identifiers](https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) (ARNs) of all the available keys. Then, if necessary, we can also check all the details about a key with a given key ID or ARN like this:
+
+{{< command >}}
+$ awslocal kms describe-key --key-id 010a4301-4205-4df8-ae52-4c2895d47326
+{{</ command >}}
+
+We can use the key now to encrypt something. Let's say, `some important stuff`:
+
+{{< command >}}
+$ awslocal kms encrypt \
+      --key-id 010a4301-4205-4df8-ae52-4c2895d47326 \
+      --plaintext "some important stuff" \
+      --output text \
+      --query CiphertextBlob \
+  | base64 --decode > my_encrypted_data
+{{</ command >}}
+
+Why do we require such a complicated command? The `Encrypt` Operation itself looks like this:
+
+{{< command >}}
+$ awslocal kms encrypt \
+      --key-id 010a4301-4205-4df8-ae52-4c2895d47326 \
+      --plaintext "some important stuff"
+{{</ command >}}
+
+And its output might be something like
+
+```sh
+{
+    "CiphertextBlob": "MDA4NDk4ZDYtMGU1ZS00NjU3LWEyZTQtMDliYTcwNDgyMmJkxw1eaLVUeUZA7bZzp+k7VAAAAAAAAAAAAAAAAAAAAAAetG9/5Znpw83/4xhwObc6Fc2B73y1ZvIigwahKopT0Q==",
+    "KeyId": "arn:aws:kms:us-east-1:000000000000:key/010a4301-4205-4df8-ae52-4c2895d47326",
+    "EncryptionAlgorithm": "SYMMETRIC_DEFAULT"
+}
+```
+
+The value of the `CiphertextBlob` field is a Base64-encoded result of encryption of our plaintext. In order to use KMS to decrypt this, we have to get rid of this Base64-encoding first. So to our `Encrypt` operation we add the `output` parameter, requesting the output to be in the text form instead of the default JSON. We also supply the `query` parameter, so the output only contains the value of the specified `CiphertextBlob` field. Then we feed the result to `base64` tool, asking it to remove the Base64 encoding, producing a binary output. We save that output to a file for future use. **Note:** Careful with the `CiphertextBlob` value for the `query` parameter - the AWS CLI doesn't check if the requested field is even if the output, so if you have a typo, the command would succeed, but the data sent to `base64` is going to be just a word `None`. Which is going to lead to unexpected results later. 
```

Comment:
```suggestion
The value of the `CiphertextBlob` field is a Base64-encoded result of the encryption of our plaintext. To use KMS to decrypt this, we must first get rid of this Base64-encoding. So to our `Encrypt` operation we add the `output` parameter, requesting the output to be in the text form instead of the default JSON. We also supply the `query` parameter, so the output only contains the value of the specified `CiphertextBlob` field. Then we feed the result to the `base64` tool, asking it to remove the Base64 encoding, producing a binary output. We save that output to a file for future use. **Note:** Careful with the `CiphertextBlob` value for the `query` parameter - the AWS CLI doesn't check if the requested field is even if the output, so if you have a typo, the command will succeed, but the data sent to `base64` is going to be just a word `None.` It will lead to unexpected results later. 
```